lib/ddtrace/profiling/scheduler.rb

```diff
@@ -90,13 +95,21 @@ def flush_and_wait
 
         # Update wait time to try to wake consistently on time.
         # Don't drop below the minimum interval.
-        self.loop_wait_time = [loop_base_interval - run_time, MIN_INTERVAL_SECONDS].max
+        self.loop_wait_time = [loop_base_interval - run_time, MINIMUM_INTERVAL_SECONDS].max
```

Comment:
totally unrelated to this PR, but I was wondering why MINIMUM_INTERVAL_SECONDS is 0? If loop_base_interval - run_time is < 0, then wouldn't this be constantly polling for profiles?

Reply:
This trips me up a bit as well.

The intent of `MINIMUM_INTERVAL_SECONDS` is just to make sure that the `loop_wait_time` is not negative. This class imports the [Loop module](https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/workers/loop.rb) (I don't quite like this design... 🤔 . This module uses this `loop_wait_time` as an input to a [`ConditionVariable#wait`](https://rubyapi.org/3.0/o/conditionvariable) call which cannot take a negative interval:

```ruby
[2] pry(main)> cv = ConditionVariable.new
=> #<Thread::ConditionVariable:0x00007f990fa28bb8>
[3] pry(main)> m = Mutex.new
=> #<Thread::Mutex:0x00007f991f925e68>
[4] pry(main)> cv.wait(m, -1)
ArgumentError: time interval must not be negative
from (pry):4:in `sleep'
```

As for the constantly polling for profiles, the intent here is that this is a bit of a "catch up" mechanism -- if a profile submission took too long, then if more than 1 minute has passed, we want to trigger the next submission immediately. 

See also https://github.com/DataDog/dd-trace-rb/pull/1017/files#r418212308 which discussed this exact behavior when this class was originally added.

Reply:
Thank you for the great answer :)